test/e2e/autotls/http01/http01_auto_tls_test.go

```diff
@@ -0,0 +1,285 @@
+// +build e2e
+
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package http01
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/kelseyhightower/envconfig"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/dns/v1"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"knative.dev/pkg/system"
+	"knative.dev/pkg/test/ingress"
+	routenames "knative.dev/serving/pkg/reconciler/route/resources/names"
+	"knative.dev/serving/test"
+	testingress "knative.dev/serving/test/conformance/ingress"
+	"knative.dev/serving/test/e2e"
+	autotls "knative.dev/serving/test/e2e/autotls"
+	v1test "knative.dev/serving/test/v1"
+)
+
+const dnsRecordDeadlineSec = 600 // 10 min
+
+type dnsRecord struct {
+	ip     string
+	domain string
+}
+
+type config struct {
+	AutoTlsDomain                 string `split_words:"true" required:"false"`
```

Comment:
well same `dns->DNS` comments apply

Reply:
From my experience, I cannot change `AutoTlsDomain` to `AutoTLSDomain` as the package `"github.com/kelseyhightower/envconfig"` won't be able to convert `AutoTLSDomain` to environment variable `AUTO_TLS_DOMAIN`. 
And it is the same for `Dns`.

Reply:
How about https://github.com/kelseyhightower/envconfig#struct-tag-support

Reply:
Good suggestion. Done